### DISCUSSION TITLE
Add a headers field to the Response data structure

### DISCUSSION THREAD

**Discussion by lewoudar** [tweepy/tweepy#1984] [Ideas]
Hello folks,
From what I know, when returning the response object, we don't have information about headers and I find it restrictive.
Ok, when instantiating the client (I'm using the v2 client) we can specify `requests.Response` as `return_type` but we lose all the advantages the `tweepy.Response` gives.
The issue here is that I want to know how many more requests I can make in the reserved window. I think I saw in another thread that there is a method we can call to have this issue, but... it is another network call.

My solution is to add `headers` information (response headers) to the response data structure. What do you think?

**Comment by Harmon758** [comment]
I'm considering changing `Response` based on this and other use cases, like #1989.
It would almost certainly be a breaking change though, so it would have to wait until a v5 release.

For now, as a workaround, with #1997, which resulted from [a conversation about this in the Tweepy Discord server](https://discord.com/channels/432685901596852224/1035394230434152448), you can use a `requests.Response` `return_type` to access the headers, before passing the JSON as a dictionary (from calling `requests.Response.json`) to `Client._construct_response`. Note, you'll also need to pass the `data_type` for any objects in `data` if you want them to be Tweepy models.

**Reply by lewoudar** [reply]
Thanks for the workaround. I will use it before the next major version is released :)
